packages/frontend/src/components/accounts/ledger/SetupLedger.js

```diff
@@ -85,10 +85,8 @@ const SetupLedger = (props) => {
                         await setKeyMeta(publicKey, { type: 'ledger' });
                         Mixpanel.track('SR-Ledger Set key meta');
 
-                        // Set custom path to localstorage
-                        if (ledgerHdPath) {
-                            setLedgerHdPath({ accountId, path: ledgerHdPath });
-                        }
+                        // Set path to localstorage
+                        setLedgerHdPath({ accountId, path: ledgerHdPath });
```

Comment:
I think we can just call this once before the if else block?
I had this in a PR yesterday(#2659),I missed it while closing and recreating the PR severally.

Reply:
We cannot because we want to add it only if adding a ledger as a recovery option was a success.

Reply:
Not sure if I'm looking at this right. But aren't we setting it before we attempt adding ledger as an option? Looks that way in the try block